tests/manager/models/test_etcd.py

```diff
@@ -0,0 +1,233 @@
+from graphene import Schema
+from graphene.test import Client
+
+from ai.backend.manager.config import SharedConfig
+from ai.backend.manager.models.gql import GraphQueryContext, Mutations, Queries
+from ai.backend.testutils.bootstrap import etcd_container  # noqa: F401
+
+CONTAINER_REGISTRY_FIELDS = """
+    container_registry {
+        hostname
+        config {
+            url
+            type
+            project
+            username
+            password
+            ssl_verify
+        }
+    }
+"""
+
+
+@pytest.fixture(scope="module")
+def client() -> Client:
+    return Client(Schema(query=Queries, mutation=Mutations, auto_camelcase=False))
+
+
+@pytest.fixture(scope="module")
+def context(etcd_container) -> GraphQueryContext:  # noqa: F811
+    shared_config = SharedConfig(
+        etcd_addr=etcd_container[1],
+        etcd_user="",
+        etcd_password="",
+        namespace="local",
+    )
+    return GraphQueryContext(
+        schema=None,  # type: ignore
+        dataloader_manager=None,  # type: ignore
+        local_config=None,  # type: ignore
+        shared_config=shared_config,  # type: ignore
+        etcd=None,  # type: ignore
+        user={"domain": "default", "role": "superadmin"},
+        access_key="AKIAIOSFODNN7EXAMPLE",
+        db=None,  # type: ignore
+        redis_stat=None,  # type: ignore
+        redis_image=None,  # type: ignore
+        redis_live=None,  # type: ignore
+        manager_status=None,  # type: ignore
+        known_slot_types=None,  # type: ignore
+        background_task_manager=None,  # type: ignore
+        storage_manager=None,  # type: ignore
+        registry=None,  # type: ignore
+        idle_checker_host=None,  # type: ignore
+    )
+
+
+@pytest.mark.dependency()
+@pytest.mark.asyncio
+async def test_create_container_registry(client: Client, context: GraphQueryContext):
+    query = """
+        mutation CreateContainerRegistry($hostname: String!, $props: CreateContainerRegistryInput!) {
+            create_container_registry(hostname: $hostname, props: $props) {
+                $CONTAINER_REGISTRY_FIELDS
+            }
+        }
+    """.replace("$CONTAINER_REGISTRY_FIELDS", CONTAINER_REGISTRY_FIELDS)
+
+    variables = {
+        "hostname": "cr.example.com",
+        "props": {
+            "url": "http://cr.example.com",
+            "type": "dockerhub",
+            "project": ["default"],
+            "username": "username",
+            "password": "password",
+            "ssl_verify": False,
+        },
+    }
+
+    response = await client.execute_async(query, variables=variables, context_value=context)
+    container_registry = response["data"]["create_container_registry"]["container_registry"]
+    assert container_registry["hostname"] == "cr.example.com"
+    assert container_registry["config"] == {
+        "url": "http://cr.example.com",
+        "type": "dockerhub",
+        "project": ["default"],
+        "username": "username",
+        "password": "*****",
```

Comment:
"*****" is defined as a constant at `ai.backend.manager.defs.PASSWORD_PLACEHOLDER`.
Let's use it.